src/main/java/com/puppycrawl/tools/checkstyle/xpath/RootNode.java

```diff
@@ -74,14 +75,20 @@ public int compareOrder(NodeInfo nodeInfo) {
     /**
      * Iterates siblings of the current node and
      * recursively creates new Xpath-nodes.
+     *
+     * @return children list
      */
-    private void createChildren() {
+    @Override
+    protected List<AbstractNode> createChildren() {
         DetailAST currentChild = detailAst;
+        final List<AbstractNode> result = new ArrayList<>();
         while (currentChild != null) {
-            final ElementNode child = new ElementNode(this, this, currentChild);
-            addChild(child);
+            final int index = result.size();
+            final ElementNode child = new ElementNode(this, this, currentChild, 0, index);
+            result.add(child);
```

Comment:
I saw this code twice in this xpath group of classes, is it reasonable to move it to util ?

Reply:
done